apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx

```diff
@@ -0,0 +1,494 @@
+    PaymentDestination.Staked
+  );
+  const [selectedValidators, setSelectedValidators] = useState<string[]>([]);
+  const [isSubmitAndSignTxLoading, setIsSubmitAndSignTxLoading] =
+    useState<boolean>(false);
+
+  const currentStep = useMemo(() => {
+    if (delegateTxStep === DelegateTxSteps.BOND_TOKENS) {
+      return '(1/3)';
+    } else if (delegateTxStep === DelegateTxSteps.SELECT_DELEGATES) {
+      return '(2/3)';
+    } else if (delegateTxStep === DelegateTxSteps.AUTHORIZE_TX) {
+      return '(3/3)';
+    }
+  }, [delegateTxStep]);
+
+  const walletAddress = useMemo(() => {
+    if (!activeAccount?.address) return '0x0';
+
+    return activeAccount.address;
+  }, [activeAccount?.address]);
+
+  const substrateAddress = useMemo(() => {
+    if (!activeAccount?.address) return '';
+
+    return convertEthereumToSubstrateAddress(activeAccount.address);
+  }, [activeAccount?.address]);
+
+  useEffect(() => {
+    const checkIfFirstTimeNominator = async () => {
+      const isFirstTimeNominator = await isNominatorFirstTimeNominator(
+        substrateAddress
+      );
+
+      setIsFirstTimeNominator(isFirstTimeNominator);
+    };
+
+    if (substrateAddress) {
+      checkIfFirstTimeNominator();
+    }
+  }, [substrateAddress]);
+
+  const { data: walletBalance } = useTokenWalletBalance(walletAddress);
+  const { data: currentPaymentDestination } =
+    usePaymentDestinationSubscription(substrateAddress);
+
+  const amountToBondError = useMemo(() => {
+    if (!walletBalance) return '';
+
+    if (Number(walletBalance.value1) === 0) {
+      return 'You have zero tTNT in your wallet!';
+    } else if (Number(walletBalance.value1) < amountToBond) {
+      return `You don't have enough tTNT in your wallet!`;
+    }
+  }, [walletBalance, amountToBond]);
+
+  const continueToSelectDelegatesStep = useMemo(() => {
+    return isFirstTimeNominator
+      ? amountToBond > 0
+      : true &&
+        !amountToBondError &&
+        paymentDestination &&
+        walletAddress !== '0x0'
+      ? true
+      : false;
+  }, [
+    amountToBond,
+    amountToBondError,
+    isFirstTimeNominator,
+    paymentDestination,
+    walletAddress,
+  ]);
+
+  const continueToAuthorizeTxStep = useMemo(() => {
+    return selectedValidators.length > 0 ? true : false;
+  }, [selectedValidators]);
+
+  const continueToSignAndSubmitTx = useMemo(() => {
+    return continueToSelectDelegatesStep && continueToAuthorizeTxStep;
+  }, [continueToSelectDelegatesStep, continueToAuthorizeTxStep]);
+
+  const closeModal = useCallback(() => {
+    setIsSubmitAndSignTxLoading(false);
+    setIsModalOpen(false);
+    setAmountToBond(0);
+    setPaymentDestination(PaymentDestination.Staked);
+    setSelectedValidators([]);
+    setDelegateTxStep(DelegateTxSteps.BOND_TOKENS);
+  }, [setIsModalOpen]);
+
+  const submitAndSignTx = useCallback(async () => {
+    setIsSubmitAndSignTxLoading(true);
+
+    try {
+      if (isFirstTimeNominator) {
```

Comment:
Lots of nesting from here. We should keep it to a max of like 2 nested conditionals if possible.